File: atcoder/abc/191d.py
import math
from decimal import Decimal
a, b, r = map(Decimal, input().split())

x=math.ceil(a-r)
y=math.floor(a+r)

cnt=0
for num in range(x,y+1):
    p=(r**2+(a-num)**2).sqrt()
    u=math.floor(b+p)
    d=math.ceil(b-p)
    cnt+=(u-d+1)

print(cnt)    

"""
import math
from decimal import Decimal

cx, cy, r = map(Decimal, input().split())

ans = 0
r2 = r ** 2
for x in range(math.ceil(cx - r), math.floor(cx + r) + 1):
    p = (r2 - (x - cx) ** 2).sqrt()

    ymin = math.ceil(cy - p)
    ymax = math.floor(cy + p)

    ans += ymax - ymin + 1

print(ans)
"""

File: atcoder/abc/191d.cpp
#include<iostream>
#include<iomanip>
#include<string>
#include<algorithm>
#include<vector>
#include<utility>
#include<tuple>
#include<map>
#include<queue>
#include<stack>
#include<deque>
#include<bitset>
#include<math.h>
#include<numeric>
using namespace std;
using ll = int64_t;
using Graph = vector<vector<int> >;
const ll M = 1000000007;

int main(){
    long double  x,y,r,p;
    cin >> x >> y >> r;
    ll low=ceil(x-r);
    ll high=floor(x+r);

    ll num=0;
    for(ll i=low;i<=high;i++){
        p=sqrt(pow(r,2)-pow((x-i),2));
        ll up=y+p;
        
        ll down=y-p;
        num+=(up-down+1);
        //cout << i << ' ' << num << endl;
    }
    cout << num << endl;
    
}